Net SDK Unit Tests/Integration Tests/EngineModuleTests.cs

```diff
@@ -143,5 +143,30 @@ public void ExecuteGetTreatment_Test_jw2_SuccessfulWithResults()
             Assert.IsTrue(result1 == "on"); //is in segment payed
             Assert.IsTrue(result2 == "off"); // default
         }
+
+        [DeploymentItem(@"Resources\splits_staging_2.json")]
+        [DeploymentItem(@"Resources\segment_payed.json")]
+        [TestMethod]
+        public void ExecuteGetTreatment_Test_jw2_SuccessfullyUsingBucketingKey()
+        {
+            //Arrange
+            ParsedSplit split = splitCache.GetSplit("test_jw2_b");
+
+            Splitter splitter = new Splitter();
+            Engine engine = new Engine(splitter);
+
+            //Act
+            //get treatment for split test_jw2_b
+            Dictionary<string, object> dict = new Dictionary<string, object>();
```

Comment:
{
      "trafficTypeName": "user",
      "name": "test_jw2_b",
      "seed": 626505653,
      "status": "ACTIVE",
      "killed": false,
      "defaultTreatment": "off",
      "changeNumber": 1470947500357,
      "conditions": [
        {
          "matcherGroup": {
            "combiner": "AND",
            "matchers": [
              {
                "keySelector": {
                  "trafficType": "user",
                  "attribute": null
                },
                "matcherType": "IN_SEGMENT",
                "negate": false,
                "userDefinedSegmentMatcherData": {
                  "segmentName": "payed"
                },
                "whitelistMatcherData": null,
                "unaryNumericMatcherData": null,
                "betweenMatcherData": null
              }
            ]
          },
          "partitions": [
            {
              "treatment": "on",
              "size": 30
            },
            {
              "treatment": "off",
              "size": 70
            }
          ]
        }
      ]
    },